nxc/protocols/ssh.py

```diff
@@ -63,124 +78,219 @@ def create_conn_obj(self):
+                stdin, stdout, stderr = self.conn.exec_command(f"echo 'sudo -s' > {pipe_stdin} && echo '{self.password}' > {pipe_stdin}")
+                # Sometime the pipe will hanging(only happen with paramiko)
+                # Can't get "whoami" or "id" result in pipe_stdout, maybe something wrong using pipe with paramiko
+                # But one thing I can confirm, is the command was executed even can't get result from pipe_stdout
+                tries = 1
+                self.logger.info(f"Copy /etc/shadow to {shadow_Backup} if pass the sudo auth")
+                while True:
+                    self.logger.info(f"Checking {shadow_Backup} if it existed")
+                    stdin, stdout, stderr = self.conn.exec_command(f'ls {shadow_Backup}')
+                    if tries >= self.args.get_output_tries:
+                        self.logger.info(f'{shadow_Backup} not existed, maybe the pipe has been hanged over, please increase the number of tries with the option "--get-output-tries" or change other method with "--sudo-check-method". If it\'s still failing maybe sudo shell is not working with current user')
+                        break
+
+                    if stderr.read().decode('utf-8'):
+                        time.sleep(2)
+                        stdin, stdout, stderr = self.conn.exec_command(f"echo 'cp /etc/shadow {shadow_Backup} && chmod 777 {shadow_Backup}' > {pipe_stdin}")
+                        tries += 1
+                    else:
+                        self.logger.info(f"{shadow_Backup} existed")
+                        self.admin_privs = True
+                        break
+                self.logger.info(f"Remove up temporary files")
+                stdin, stdout, stderr = self.conn.exec_command(f"rm -rf {shadow_Backup} {pipe_stdin} {pipe_stdout}")
+            else:
+                self.logger.error("Command: 'mkfifo' unavailable, running command with 'sudo' failed")
+                return
 
     def plaintext_login(self, username, password, private_key=None):
+        self.username = username
+        self.password = password
+        private_key = ""
+        stdout = None
+        stderr = None
         try:
             if self.args.key_file or private_key:
-                if private_key:
-                    pkey = paramiko.RSAKey.from_private_key(StringIO(private_key))
-                else:
-                    pkey = paramiko.RSAKey.from_private_key_file(self.args.key_file)
-
                 self.logger.debug(f"Logging in with key")
-                self.conn.connect(
-                    self.host,
-                    port=self.args.port,
-                    username=username,
-                    passphrase=password if password != "" else None,
-                    pkey=pkey,
-                    look_for_keys=False,
-                    allow_agent=False,
+
+                if self.args.key_file:
+                    with open(self.args.key_file, 'r') as f:
```

Comment:
Please also make sure to use `'` only inside strings. Outside quotes should always be double quotes `"`

Reply:
Sure!

Reply:
Fixed!